src/brains2/src/sim/sim_node.cpp

```diff
@@ -0,0 +1,499 @@
+// Copyright (c) 2024. Tudor Oancea, Matteo Berthet
+#include <cmath>
+#include <fstream>
+#include <tuple>
+#include <unordered_map>
+#include "ament_index_cpp/get_package_share_directory.hpp"
+#include "brains2/common/cone_color.hpp"
+#include "brains2/common/marker_color.hpp"
+#include "brains2/common/math.hpp"
+#include "brains2/common/tracks.hpp"
+#include "brains2/external/icecream.hpp"
+#include "brains2/external/optional.hpp"
+#include "brains2/msg/acceleration.hpp"
+#include "brains2/msg/controls.hpp"
+#include "brains2/msg/pose.hpp"
+#include "brains2/msg/velocity.hpp"
+#include "brains2/sim/sim.hpp"
+#include "diagnostic_msgs/msg/diagnostic_array.hpp"
+#include "diagnostic_msgs/msg/diagnostic_status.hpp"
+#include "diagnostic_msgs/msg/key_value.hpp"
+#include "Eigen/Dense"
+#include "geometry_msgs/msg/pose.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "geometry_msgs/msg/quaternion.hpp"
+#include "geometry_msgs/msg/transform_stamped.hpp"
+#include "geometry_msgs/msg/twist.hpp"
+#include "geometry_msgs/msg/twist_stamped.hpp"
+#include "geometry_msgs/msg/vector3.hpp"
+#include "nlohmann/json.hpp"
+#include "rcl_interfaces/msg/set_parameters_result.hpp"
+#include "rclcpp/publisher.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/color_rgba.hpp"
+#include "std_msgs/msg/float64.hpp"
+#include "std_msgs/msg/header.hpp"
+#include "std_srvs/srv/empty.hpp"
+#include "tf2/LinearMath/Quaternion.h"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+#include "tf2_ros/transform_broadcaster.h"
+#include "visualization_msgs/msg/marker.hpp"
+#include "visualization_msgs/msg/marker_array.hpp"
+#include "yaml-cpp/yaml.h"
```

Comment:
A lot of useless imports

Reply:
Some of them actually are (e.g. nlohman/json) but not all of them, even though clangd indicates they are. 

My theory is that it treats them as 'useless' (by indicating that 'included header is not used directly') if the used symbols are not declared in this very header file, and this even though they are declared in another header included in the first. This is the case with ros message types, where we usually include a single header (e.g. `header.hpp`) that includes several others (e.g. `header__struct.hpp`, where the `Header` class is actually defined).

This is pure speculation though, and should be confirmed by researching the actual way clangd works

Reply:
Ah yes ROS 2 messages don't seem to be well considered. But at least these are useless I don't know if it matters tho.
 
```cpp
#include <cmath>
#include <fstream>
#include "diagnostic_msgs/msg/key_value.hpp"
#include "geometry_msgs/msg/vector3.hpp"
#include "nlohmann/json.hpp"
#include "rcl_interfaces/msg/set_parameters_result.hpp"
#include "tf2/LinearMath/Quaternion.h"
#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
```

Reply:
I removed them